codex/validation.nim

```diff
@@ -67,18 +72,22 @@ proc subscribeSlotFilled(validation: Validation) {.async.} =
   proc onSlotFilled(requestId: RequestId, slotIndex: UInt256) =
     let slotId = slotId(requestId, slotIndex)
     if validation.shouldValidateSlot(slotId):
-      trace "Adding slot", slotId
+      trace "Adding slot", slotId, groups = validation.config.groups,
+        groupIndex = validation.config.groupIndex
       validation.slots.incl(slotId)
   let subscription = await validation.market.subscribeSlotFilled(onSlotFilled)
   validation.subscriptions.add(subscription)
 
 proc removeSlotsThatHaveEnded(validation: Validation) {.async.} =
+  logScope:
+    groups = validation.config.groups
+    groupIndex = validation.config.groupIndex
   var ended: HashSet[SlotId]
   let slots = validation.slots
   for slotId in slots:
     let state = await validation.market.slotState(slotId)
     if state != SlotState.Filled:
-      trace "Removing slot", slotId
+      trace "Removing slot", slotId, slotState = state
```

Comment:
This is in function `removeSlotsThatHaveEnded`. Saying "unfilled" in this case feels like talking about a slot that has never been filled while the opposite is true. Wouldn't you agree that this might be confusing?